http/auth.go

```diff
@@ -51,31 +51,33 @@ func reCaptcha(host, secret, response string) (bool, error) {
 
 // authHandler processes the authentication for the user.
 func authHandler(c *fb.Context, w http.ResponseWriter, r *http.Request) (int, error) {
-	// NoAuth instances shouldn't call this method.
 	if c.NoAuth {
+		// NoAuth instances shouldn't call this method.
 		return 0, nil
-	}
-
-	// Receive the credentials from the request and unmarshal them.
-	var cred cred
-	if r.Body == nil {
-		return http.StatusForbidden, nil
-	}
-
-	err := json.NewDecoder(r.Body).Decode(&cred)
-	if err != nil {
-		return http.StatusForbidden, nil
-	}
+	} else if c.AuthMethod == "proxy" {
```

Comment:
I think this is an unnecessary branch. The previous `if` contains a return statement, so there is no need for an `else` here. The flow will automatically execute `if c.AuthMethod == "proxy"` after evaluating `c.NoAuth` to false. Please, remove `} else`.